silx/io/dictdump.py

```diff
@@ -37,41 +37,39 @@
 from .utils import is_group, is_link, is_softlink, is_externallink
 from .utils import is_file as is_h5_file_like
 from .utils import open as h5open
+from .utils import h5py_read_dataset
+from .utils import H5pyAttributesReadWrapper
 
 __authors__ = ["P. Knobel"]
 __license__ = "MIT"
 __date__ = "17/07/2018"
 
 logger = logging.getLogger(__name__)
 
-string_types = (basestring,) if sys.version_info[0] == 2 else (str,)    # noqa
+try:
+    vlen_utf8 = h5py.special_dtype(vlen=unicode if sys.version_info[0] == 2 else str)
+    vlen_bytes = h5py.special_dtype(vlen=bytes)
+except AttributeError:
+    # h5py does not support variable-length types
+    vlen_utf8 = vlen_bytes = None
```

Comment:
vlen seems supported since some time.
What about setting a minimum version of `h5py`? I opened an issue for it: https://github.com/silx-kit/silx/issues/3246

Reply:
BTW, this can be cleaned-up with #3246.

Reply:
Hmmm I wonder why I did that. I tried down to 2.8 but 2.8 already has vlen. I'll remove it.

Reply:
Removed it.